src/Clapi/Valuespace.hs

```diff
@@ -340,7 +340,9 @@ validateChildKeyTypes getType' np n expectedTypes =
     expectedTypeMap = Map.fromList $
         zip keys (zip expectedTypes (getChildPaths np n))
     failTypes bad = when (not . null $ bad) $ fail $
-        printf "bad child types %s" (show bad)
+        printf "bad child types for %s:%s" (show np) (concat $ map fmtBct $ Map.assocs bad)
```

Comment:
[`concatMap`](http://hackage.haskell.org/package/base-4.10.0.0/docs/Data-Foldable.html#v:concatMap) is a thing - I think it might be more efficient in some circumstances, but my memory is foggy!